File: Find_Peak_Element.cpp
class Solution {
public:
    /**
     * @param A: An integers array.
     * @return: return any of peek positions.
     */
    int findPeak(vector<int> A) {
        // write your code here
        int s = 0;
        int e = A.size() - 1;
        while (s <= e){
            int m = s + (e - s) / 2;
            if (A[m - 1] <= A[m]){
                s = m + 1;
            }else{
                e = m - 1;
            }
        }
        return s - 1;
    }
};

File: Find Peak Element.py
class Solution:
    #@param A: An integers list.
    #@return: return any of peek positions.
    def findPeak(self, A):
        # write your code here
        
        left, right = 0, len(A) - 1
        
        while left + 1 < right:
            mid = left + (right - left) / 2
            if A[mid] < A[mid + 1]:
                left = mid
            else:
                right = mid
        return left if A[left] > A[right] else right